SU2_CFD/src/solvers/CEulerSolver.cpp

```diff
@@ -3694,27 +3381,21 @@ void CEulerSolver::Source_Residual(CGeometry *geometry, CSolver **solver_contain
       numerics->SetVolume(geometry->node[iPoint]->GetVolume());
 
       /*--- Compute the rotating frame source residual ---*/
-      numerics->ComputeResidual(Residual, Jacobian_i, config);
+      auto residual = numerics->ComputeResidual(config);
 
       /*--- Add the source residual to the total ---*/
-      LinSysRes.AddBlock(iPoint, Residual);
+      LinSysRes.AddBlock(iPoint, residual);
 
       /*--- Add the implicit Jacobian contribution ---*/
-      if (implicit) Jacobian.AddBlock(iPoint, iPoint, Jacobian_i);
+      if (implicit) Jacobian.AddBlock2Diag(iPoint, residual.jacobian_i);
```

Comment:
You will see that in boundary conditions and Source_Residual loops this new method is used to write directly to the diagonal of the matrix, `AddBlock2Diag` uses the lookup array to access the diagonal instead of the exhaustive search in `AddBlock`, there is no loss of generality since that lookup array is always created.